panic: runtime error: index out of range

Running a simple `Render` method benchmarks in parallel result to: `panic: runtime error: index out of range`.

The code:

``` go
func BenchmarkRenderParallel(b *testing.B) {
    b.RunParallel(func(pb *testing.PB) {
        for pb.Next() {
            err := Render("XXX", "YYY", ColorBlue, ioutil.Discard)
            if err != nil {
                b.Fatal(err)
            }
        }
    })
}
```

```
goroutine 23 [running]:
panic(0x23af60, 0xc820012060)
    /usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
github.com/golang/freetype/truetype.(*GlyphBuf).addPhantomsAndScale(0xc8200fab00, 0x0, 0x0, 0x101)
    /Users/varankinv/Workspace/golang/src/github.com/golang/freetype/truetype/glyph.go:502 +0x673
github.com/golang/freetype/truetype.(*GlyphBuf).load(0xc8200fab00, 0x1003b00000000, 0x0, 0x0)
    /Users/varankinv/Workspace/golang/src/github.com/golang/freetype/truetype/glyph.go:229 +0x8bb
github.com/golang/freetype/truetype.(*GlyphBuf).Load(0xc8200fab00, 0xc8200f0000, 0x3b000002c0, 0x2, 0x0, 0x0)
    /Users/varankinv/Workspace/golang/src/github.com/golang/freetype/truetype/glyph.go:102 +0x135
github.com/golang/freetype/truetype.(*face).GlyphAdvance(0xc8200f8000, 0x58, 0x0)
    /Users/varankinv/Workspace/golang/src/github.com/golang/freetype/truetype/face.go:333 +0x6d
golang.org/x/image/font.(*Drawer).MeasureString(0xc8200147c0, 0x284540, 0x3, 0x0)
    /Users/varankinv/Workspace/golang/src/golang.org/x/image/font/font.go:135 +0x112
github.com/narqo/go-badge.(*badgeDrawer).measureString(0xc8201540a0, 0x284540, 0x3, 0x0)
    /Users/varankinv/Workspace/golang/src/github.com/narqo/go-badge/badge.go:60 +0x3b
github.com/narqo/go-badge.(*badgeDrawer).Render(0xc8201540a0, 0x284540, 0x3, 0x2845a8, 0x3, 0x284eb8, 0x4, 0x49e4c0, 0xc820012518, 0x0, ...)
    /Users/varankinv/Workspace/golang/src/github.com/narqo/go-badge/badge.go:38 +0x5c
github.com/narqo/go-badge.Render(0x284540, 0x3, 0x2845a8, 0x3, 0x284eb8, 0x4, 0x49e4c0, 0xc820012518, 0x0, 0x0)
    /Users/varankinv/Workspace/golang/src/github.com/narqo/go-badge/badge.go:67 +0x95
github.com/narqo/go-badge.BenchmarkRenderParallel.func1(0xc8201440e0)
    /Users/varankinv/Workspace/golang/src/github.com/narqo/go-badge/badge_test.go:23 +0xaa
testing.(*B).RunParallel.func1(0xc8201540b0, 0xc820154018, 0xc820154010, 0xc820178000, 0xc820154000)
    /usr/local/Cellar/go/1.6/libexec/src/testing/benchmark.go:449 +0xa6
created by testing.(*B).RunParallel
    /usr/local/Cellar/go/1.6/libexec/src/testing/benchmark.go:450 +0x178
panic: runtime error: slice bounds out of range
```

--- a/badge.go
+++ b/badge.go
@@ -3,6 +3,7 @@ package badge
 import (
 	"html/template"
 	"io"
+	"sync"
 
 	"github.com/golang/freetype/truetype"
 	"github.com/narqo/go-badge/fonts"
@@ -30,13 +31,16 @@ func (b bounds) Dx() float64 {
 }
 
 type badgeDrawer struct {
-	fd   *font.Drawer
-	tmpl *template.Template
+	fd    *font.Drawer
+	tmpl  *template.Template
+	mutex *sync.Mutex
 }
 
 func (d *badgeDrawer) Render(subject, status string, color Color, w io.Writer) error {
+	d.mutex.Lock()
 	subjectDx := d.measureString(subject)
 	statusDx := d.measureString(status)
+	d.mutex.Unlock()
 
 	bdg := badge{
 		Subject: subject,
@@ -78,6 +82,7 @@ func init() {
 	drawer = &badgeDrawer{
 		fd:   mustNewFontDrawer(fontsize, dpi),
 		tmpl: template.Must(template.New("flat-template").Parse(flatTemplate)),
+		mutex: &sync.Mutex{},
 	}
 }
 

--- a/badge_test.go
+++ b/badge_test.go
@@ -5,7 +5,7 @@ import (
 	"testing"
 )
 
-func BenchmarkRender(b *testing.B) {
+func BenchmarkRenderSeq(b *testing.B) {
 	// warm up
 	Render("XXX", "YYY", ColorBlue, ioutil.Discard)
 	b.ResetTimer()
@@ -15,4 +15,15 @@ func BenchmarkRender(b *testing.B) {
 			b.Fatal(err)
 		}
 	}
+}
+
+func BenchmarkRenderParallel(b *testing.B) {
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			err := Render("XXX", "YYY", ColorBlue, ioutil.Discard)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
 }

